src/site/topic20.rst

```diff
@@ -97,6 +97,43 @@ Binary Search
 Iterative
 ---------
 
+* Below is a generic implementation of an iterative binary search on an array of type ``T``
+    * Take note that ``T`` or one of their superclasses must extend ``Comparable``
+    * This is because the elements must be ordered
+
+.. code-block:: java
+    :linenos:
+
+    public static <T extends Comparable<? super T>> int iterativeBinarySearch(T needle, T[] haystack) {
+        int lowIndex = 0;
+        int highIndex = haystack.length;
+        int midpoint = (highIndex - lowIndex) / 2;
+
+        while (lowIndex < highIndex) {
+            if (haystack[midpoint].equals(needle)) {
+                return midpoint;
+            } else if (haystack[midpoint].compareTo(needle) > 0) {
+                highIndex = midpoint - 1;
+                midpoint = lowIndex + (highIndex - lowIndex) / 2;
+            } else {
+                lowIndex = midpoint + 1;
+                midpoint = lowIndex + (highIndex - lowIndex) / 2;
+            }
+        }
+        return -1;
+    }
+
+* This may look complicated, but again, take your time
+
+* Here is what's happening
```

Comment:
May want to also explain why the `while` condition is what it is. Something like "repeat until we've exhaused the search space" -- but iunno how to clarify "exhaused the search space" any better. Sounds very mathy.